### ISSUE TITLE
Add plotting and logging for hypergrid scripts and tutorials

### ISSUE THREAD

**Issue by saleml** [GFNOrg/torchgfn#226]
Seeing the training loop work is definitely a good sign, but is not enough. There could be many silent errors:
- the loss could explode
- even with a good loss, the learned distribution could be getting further from the target distribution

There is also no way to visualize the learned distributions, for the environment we use throughout our examples and tutorials, hypergrid.

I suggest we take inspiration from this [notebook](https://github.com/GFNOrg/GFN_vs_HVI/blob/master/grid/wandb_and_plots/download_data_wandb.ipynb) to implement a few utility functions, and end all hypergrid related scripts with plots.

What do you think about this @josephdviviano ?

**Comment by josephdviviano** [comment]
I think this is a great idea and you did a great job on the PR :)
